proxy/envoy/testdata/cds-ssl-context-optin.json.golden

```diff
@@ -0,0 +1,206 @@
+    "service_name": "world.default.svc.cluster.local|mongo",
+    "connect_timeout_ms": 1000,
+    "type": "sds",
+    "lb_type": "round_robin"
+   },
+   {
+    "name": "out.242bc3028e0f3fe0682e6d972e167ab415b2321d",
+    "connect_timeout_ms": 1000,
+    "type": "strict_dns",
+    "lb_type": "round_robin",
+    "hosts": [
+     {
+      "url": "tcp://istio-egress.istio-system:8888"
+     }
+    ]
+   },
+   {
+    "name": "out.5898aa4379cc19c8f1bb3b7915ee8e0e32ddc6a6",
+    "service_name": "world.default.svc.cluster.local|custom",
+    "connect_timeout_ms": 1000,
+    "type": "sds",
+    "lb_type": "round_robin"
+   },
+   {
+    "name": "out.6fc71bd0fc240b9cf1b3437a82c18a5c97d40890",
+    "service_name": "hello.default.svc.cluster.local|mongo",
+    "connect_timeout_ms": 1000,
+    "type": "sds",
+    "lb_type": "round_robin"
+   },
+   {
+    "name": "out.81c187d71467b1608736c57bb0734f9ef9b68f7d",
+    "service_name": "hello.default.svc.cluster.local|http-status",
+    "connect_timeout_ms": 1000,
+    "type": "sds",
+    "lb_type": "round_robin"
+   },
+   {
+    "name": "out.a2263b90cd24e500e9ed95f79ae47eabdc77dc74",
+    "service_name": "world.default.svc.cluster.local|http",
+    "connect_timeout_ms": 1000,
+    "type": "sds",
+    "lb_type": "round_robin",
+    "ssl_context": {
+     "cert_chain_file": "/etc/certs/cert-chain.pem",
+     "private_key_file": "/etc/certs/key.pem",
+     "ca_cert_file": "/etc/certs/root-cert.pem",
+     "verify_subject_alt_name": [
+      "spiffe://cluster.local/ns/default/sa/serviceaccount1",
+      "spiffe://cluster.local/ns/default/sa/serviceaccount2"
+     ]
+    }
+   },
+   {
+    "name": "out.ad08e00d6d07a7cdbdbacfc83fef3fd62aaa4c26",
+    "service_name": "hello.default.svc.cluster.local|redis",
+    "connect_timeout_ms": 1000,
+    "type": "sds",
+    "lb_type": "round_robin"
+   },
+   {
+    "name": "out.ae8d3361601f8293abe6ac5e4d807124612cf42e",
+    "connect_timeout_ms": 1000,
+    "type": "strict_dns",
+    "lb_type": "round_robin",
+    "hosts": [
+     {
+      "url": "tcp://istio-egress.istio-system:8888"
+     }
+    ]
+   },
+   {
+    "name": "out.bde94496eb59ec2ed5b81392a1d32377960660b8",
+    "service_name": "world.default.svc.cluster.local|http-status",
+    "connect_timeout_ms": 1000,
+    "type": "sds",
+    "lb_type": "round_robin"
+   },
+   {
+    "name": "out.de6d66d4dd5f542e5f61882eb466189eb68ebe88",
+    "service_name": "hello.default.svc.cluster.local|custom",
+    "connect_timeout_ms": 1000,
+    "type": "sds",
+    "lb_type": "round_robin"
+   },
+   {
+    "name": "out.e5c9564b7c4dbb0355a4f740e9d29277ccca97cd",
+    "service_name": "hello.default.svc.cluster.local|http",
+    "connect_timeout_ms": 1000,
+    "type": "sds",
+    "lb_type": "round_robin",
+    "ssl_context": {
+     "cert_chain_file": "/etc/certs/cert-chain.pem",
+     "private_key_file": "/etc/certs/key.pem",
+     "ca_cert_file": "/etc/certs/root-cert.pem",
+     "verify_subject_alt_name": []
```

Comment:
why is this empty?

Reply:
I guess the mock setup doesn't have service account for this service. 
https://github.com/istio/pilot/blob/master/test/mock/service.go#L257

This ssl_context is the same as the one in the existing proxy/envoy/testdata/cds-ssl-context.json.golden